File: backend/src/models/Product.js
const mongoose = require("mongoose");

const productColorSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    hex: { type: String, required: true, trim: true },
  },
  { _id: false }
);

const productSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    price: { type: Number, required: true, min: 0 },

    status: {
      type: String,
      enum: ["active", "inactive"],
      default: "active",
      required: true,
    },

    category: { type: String, required: true, trim: true },

    image: { type: String, required: true, trim: true },

    // OPCIONAL (como pediste)
    imageAlt: { type: String, trim: true },

    colors: { type: [productColorSchema], default: [] },

    sizes: { type: [String], default: [] },

    description: { type: String, required: true, trim: true },

    quantity: { type: Number, required: true, min: 0, default: 0 },

    createdBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Product", productSchema);

File: backend_python/app/schemas/product.py
from pydantic import BaseModel
from typing import List, Literal, Optional


class ProductColor(BaseModel):
    name: str
    hex: str


class ProductCreate(BaseModel):
    name: str
    price: float
    status: Literal["active", "inactive"] = "active"
    category: str
    image: str
    imageAlt: Optional[str] = None
    colors: List[ProductColor] = []
    sizes: List[str] = []
    description: str
    quantity: int = 0


class ProductUpdate(BaseModel):
    name: Optional[str] = None
    price: Optional[float] = None
    status: Optional[Literal["active", "inactive"]] = None
    category: Optional[str] = None
    image: Optional[str] = None
    imageAlt: Optional[str] = None
    colors: Optional[List[ProductColor]] = None
    sizes: Optional[List[str]] = None
    description: Optional[str] = None
    quantity: Optional[int] = None
